aptos-move/aptos-vm/src/transaction_arg_validation.rs

```diff
@@ -56,10 +69,15 @@ pub(crate) fn validate_combine_signer_and_txn_args<S: MoveResolverExt>(
         }
     }
     // validate all non_signer params
-    for ty in func.parameters[signer_param_cnt..].iter() {
-        if !is_valid_txn_arg(session, ty) {
+    let mut needs_validation = vec![];
```

Comment:
should we change this to a `Set`?
I run under the assumption a worst case scenario here cannot be that big anyway (this is a function of the number of arguments to an entry), but maybe we should do it for "safety"?

Reply:
I think it is fine